File: Python/ugly_number.py
# https://leetcode.com/problems/ugly-number/description/
from utils.examination import Examiner, Testcase


class Solution:
    def isUgly(self, n: int) -> bool:
        if 0 >= n:
            return False
        while 1 < n:
            if 0 == n % 2:
                n = int(n / 2)
                continue
            if 0 == n % 3:
                n = int(n / 3)
                continue
            if 0 == n % 5:
                n = int(n / 5)
                continue
            return False
        return True

def examine()->None:
    Examiner(Solution(), 'isUgly').run((
        Testcase((6,), True),
        Testcase((1,), True),
        Testcase((14,), False),
    ))

File: TypeScript/solutions/ugly-number.ts
// https://leetcode.com/problems/ugly-number/description/
import { Examiner, Testcase } from '../utils';


function isUgly(n: number): boolean {
    if (n <= 0) return false;
    while (n > 1) {
        if (n % 2 === 0) { n = Math.floor(n / 2); continue; }
        if (n % 3 === 0) { n = Math.floor(n / 3); continue; }
        if (n % 5 === 0) { n = Math.floor(n / 5); continue; }
        return false;
    }
    return true;
}

export function examine(): void {
    new Examiner(isUgly).run([
        new Testcase([6], true),
        new Testcase([1], true),
        new Testcase([14], false),
    ]);
}
